src/synapse/api.py

```diff
@@ -223,3 +234,225 @@ def get_version() -> str:
+
+    Args:
+        user: the user on behalf of whom you want to request the access token.
+        server: to be used to create the user id. User ID example: @user:server.com.
+        admin_access_token: a server admin access token to be used for the request.
+
+    Returns:
+        Access token.
+
+    Raises:
+        GetAccessTokenError: if there was an error while getting access token.
+    """
+    authorization_token = f"Bearer {admin_access_token}"
+    headers = {"Authorization": authorization_token}
+    # @user:server.com
+    user_id = f"@{user.username}:{server}"
+    res = _do_request("POST", f"{LOGIN_URL}/{user_id}/login", headers=headers)
+    try:
+        res_access_token = res.json()["access_token"]
+    except (requests.exceptions.JSONDecodeError, KeyError, TypeError) as exc:
+        logger.exception("Failed to decode access_token: %r. Received: %s", exc, res.text)
+        raise GetAccessTokenError(str(exc)) from exc
+    return res_access_token
+
+
+def override_rate_limit(user: User, admin_access_token: str, charm_state: CharmState) -> None:
+    """Override user's rate limit.
+
+    Args:
+        user: user to be used for requesting access token.
+        admin_access_token: server admin access token to be used.
+        charm_state: Instance of CharmState.
+    """
+    server_name = charm_state.server_name
+    rate_limit_url = (
+        f"{SYNAPSE_URL}/_synapse/admin/v1/users/"
+        f"@{user.username}:{server_name}/override_ratelimit"
+    )
+    authorization_token = f"Bearer {admin_access_token}"
+    headers = {"Authorization": authorization_token}
+    _do_request("DELETE", rate_limit_url, headers=headers)
+
+
+def get_room_id(
+    room_name: str,
+    admin_access_token: str,
+) -> str:
+    """Get room id.
+
+    Args:
+        room_name: room name.
+        admin_access_token: server admin access token to be used.
+
+    Returns:
+        The room id.
+
+    Raises:
+        GetRoomIDError: if there was an error while getting room id.
+        RoomNotFoundError: if the room was not found.
+    """
+    authorization_token = f"Bearer {admin_access_token}"
+    headers = {"Authorization": authorization_token}
+    res = _do_request("GET", f"{LIST_ROOMS_URL}?search_term={room_name}", headers=headers)
+    try:
+        rooms = res.json()["rooms"]
+        total = len(rooms)
+        if total > 1:
+            logger.warning(
+                "%s rooms named with term %s found, checking for exact name.", total, room_name
+            )
+        for room in rooms:
+            if str(room["name"]).upper() == room_name.upper():
+                return room["room_id"]
+    except (requests.exceptions.JSONDecodeError, TypeError, KeyError) as exc:
+        logger.exception("Failed to decode rooms: %r. Received: %s", exc, res.text)
+        raise GetRoomIDError(str(exc)) from exc
+
+    raise RoomNotFoundError(f"Room {room_name} not found. Please, verify if the room exists.")
```

Comment:
Why not returning None instead?

Reply:
Hmm...good point. I don't have an answer for that, I was following the flow of creating exceptions, perhaps.
I'll change it.

Reply:
Changed, thanks :)